mmda/parsers/parser.py

```diff
@@ -12,21 +12,20 @@
 from pdf2image import convert_from_path
 
 from mmda.types.document import Document
-from mmda.types.image import Image
 
 
 class BaseParser:
 
     @abstractmethod
-    def parse(
-        self, infile: str, outdir: Optional[str] = None, outfname: Optional[str] = None, load_images=True
-    ) -> Union[str, Document]:
+    def parse(self, input_pdf_path: str, output_json_path: Optional[str] = None,
+              tempdir: Optional[str] = None, load_images: bool = False) -> Union[str, Document]:
         """This is the main entrance point for using the PDF parsers. For a
         given PDF file, this method will return a Document object.
         """
 
+    # TODO[kylel] - this doesnt quite work; convert_from_path() doesnt use our Image w patches; it uses PIL.Image
+    # TODO[kylel] - serialization?
```

Comment:
Yes, but fundamentally the output type is just an inherited class based on `PIL.Image` that is designed for handling loading images of different types. I had this initial confusion as you did, but I checked and found they are basically the same. See here: http://code.nabla.net/doc/PIL/api/PIL/PpmImagePlugin/PIL.PpmImagePlugin.PpmImageFile.html#PIL.PpmImagePlugin.PpmImageFile (OK this is an old documentation, but it shows the inheritance graph).

And if you try `image.copy()`, then the output is just `PIL.Image`. 

<img width="315" alt="image" src="https://user-images.githubusercontent.com/22512825/128657586-e0e86b32-6e33-4ee7-8329-c49e03c6df1e.png">

I don't think this is the optimal solution right now -- but we might keep it just for the time being..